### PR TITLE
Add test to ensure we only sync users of the current blog in multisite

### PR DESCRIPTION
**Pull request by gravityrail**
Ensure that we sync all users who are members of the current site, and that we delete synced users when they stop being members of the current site.

cc @lezama

### CODE REVIEW

File: tests/php/sync/test_class.jetpack-sync-users.php [hunk 1]
```diff
@@ -212,7 +212,59 @@ public function test_user_remove_all_capability_is_synced() {
...
+		remove_user_from_blog( $other_blog_user_id, $original_blog_id );
+
+		$this->client->do_sync();
+
+		$this->assertNull( $this->server_replica_storage->get_user( $other_blog_user_id ) );
+	}
+
+	public function test_syncs_users_added_to_multisite() {
+		global $wpdb;
+
```
Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-users.php]: This test should be multisite only as well.

**Comment by lezama** [comment]
Thanks for tackling this one Dan!

File: sync/class.jetpack-sync-client.php [hunk 2]
```diff
@@ -133,8 +133,11 @@ private function init() {
 		// users
 		add_action( 'user_register', array( $this, 'save_user_handler' ) );
 		add_action( 'profile_update', array( $this, 'save_user_handler' ), 10, 2 );
+		add_action( 'add_user_to_blog', array( $this, 'save_user_handler' ) );
```
Comment (lezama) [inline sync/class.jetpack-sync-client.php]: nice

File: tests/php/sync/test_class.jetpack-sync-users.php [hunk 3]
```diff
@@ -212,7 +212,63 @@ public function test_user_remove_all_capability_is_synced() {
...
+		$other_blog_id = wpmu_create_blog( 'foo.com', '', "My Blog", $this->user_id );
+		$wpdb->suppress_errors( $suppress );
+
+		$other_blog_user_id = $this->factory->user->create();
+		add_user_to_blog( $other_blog_id, $other_blog_user_id, 'administrator' );
+		remove_user_from_blog( $other_blog_user_id, $original_blog_id );
+
+		$this->client->do_sync();
+
+		$this->assertNull( $this->server_replica_storage->get_user( $other_blog_user_id ) );
```
Comment (lezama) [inline tests/php/sync/test_class.jetpack-sync-users.php]: 👍

**Comment by lezama** [comment]
Looks great! Maybe we need to add one test to check that we don't sync them on "full_sync"?

**Comment by gravityrail** [comment]
@lezama - done.

**Pull request merged**